dist/aws/handler.py

```diff
@@ -1,144 +1,215 @@
-        output_filename = file_key.replace(".json", "." + pipeline["output"]["format"]).replace(
-            settings_trigger["FUNCTION_S3_INPUT"], settings_trigger["FUNCTION_S3_OUTPUT"]
+        # create output filename
+        pipeline["output"]["filename"] = file_key.replace(".json", "." + pipeline["output"]["format"]).replace(
+            pipeline["settings"]["FUNCTION_S3_INPUT"], pipeline["settings"]["FUNCTION_S3_OUTPUT"]
         )
 
-        try:
-            s3.head_object(Bucket=bucket, Key=output_filename)
+        if not pipeline["settings"]["FUNCTION_FORCE_COMPUTE"]:
 
-            # Object exists, so we don't have to recompute
-            # TODO: the "force_compute" parameter will never work as is written in aws.py
-            if not pipeline.get("force_compute", False):
-                cached = True
+            # get configured s3 bucket to check for cache
+            bucket = pipeline["settings"]["S3_BUCKET_NAME"]
 
-        except botocore.exceptions.ClientError:
-            pass
+            # We can return if there is a valid cached output to save compute time.
+            try:
+                s3.head_object(Bucket=bucket, Key=pipeline["output"]["filename"])
+                return None
 
-        if cached:
-            return
+            # throws ClientError if no file is found
+            except botocore.exceptions.ClientError:
+                pass
+
+        # return pipeline definition
+        return pipeline
+
+    elif trigger == "APIGateway":
+        print("Triggered from API Gateway")
+
+        pipeline = default_pipeline()
+        pipeline["url"] = event["queryStringParameters"]
+        pipeline["params"] = urllib.parse_qs(urllib.urlparse(pipeline["url"]).query)
+
+        # make all params lowercase
+        pipeline["params"] = [param.lower() for param in pipeline["params"]]
+
+        # look for specific parameter definitions in query parameters
+        for param in pipeline["params"]:
+            
+            # handle SETTINGS in query parameters
+            # TODO: shouldn't this merge trigger-provided settings with default?
+            if param == "settings":
+                pipeline["settings"] = pipeline["params"][param]
+
+            # handle OUTPUT in query parameters
+            elif param == "output":
+                pipeline["output"] = pipeline["params"][param]
+
+            # handle FORMAT in query parameters
+            elif param == "format":
+                pipeline["output"][param] = pipeline["params"][param].split("/")[-1]
+                # handle image returns
+                if pipeline["output"][param] in ["png", "jpg", "jpeg"]:
+                    pipeline["output"]["format_kwargs"]["return_base64"] = True
+
+        return pipeline
 
-    # TODO: handle "invoke" and "APIGateway" triggers explicitly
     else:
-        print("Not triggered by S3")
+        raise Exception("Unsupported trigger")
 
-        url = event["queryStringParameters"]
```

Comment:
It looks like it was in the porting of this block that we lost the isinstance check.